### PR TITLE
(HTML-safe) inline markdown for publisher description.

### LINKED ISSUE #3527: Allow inline markdown syntax for publisher description
https://pub.dev/documentation/markdown/latest/markdown/markdownToHtml.html

Has an option `inlineOnly: true`, we probably have to test exactly what it does..
Maybe we need to customize the syntaxes enabled through https://pub.dev/documentation/markdown/latest/markdown/ExtensionSet-class.html.

But if we can avoid any of the block syntaxes that would be nice.

I would also like to avoid allowing images.

### PR DESCRIPTION
**Pull request by isoos**
- fixes #3527
- only `strong` and `em` elements are allowed/exposed

### CODE REVIEW

File: app/lib/shared/markdown.dart [hunk 1]
```diff
@@ -81,7 +92,23 @@ String _renderSafeHtml(List<m.Node> nodes) {
       return _whitelistedClassNames.contains(cn);
     },
   );
-  return html + '\n';
+  return inlineOnly ? html : '$html\n';
+}
+
+final _safeInlineTags = <String>{'em', 'strong'};
```
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/shared/markdown.dart:98]: hmm, I can see that in search results we'll want to render inline without links... and maybe even want to strip bold.. 

But for side-bar and publisher description inline links seems okay.

File: app/lib/shared/markdown.dart [hunk 2]
```diff
@@ -24,27 +24,35 @@ const _whitelistedClassNames = <String>[
...
 
 /// Parses markdown [source].
-List<m.Node> _parseMarkdownSource(String source) {
+List<m.Node> _parseMarkdownSource(String source, bool inlineOnly) {
   if (source == null) return null;
   final document = m.Document(
       extensionSet: m.ExtensionSet.gitHubWeb,
       blockSyntaxes: m.ExtensionSet.gitHubWeb.blockSyntaxes);
+  if (inlineOnly) {
+    return document.parseInline(source);
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/shared/markdown.dart:54]: I don't think that it is needed, but to make sure, I'll add a test for it.

File: app/lib/shared/markdown.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/shared/markdown.dart:98]: Good question. I'd probably start with a smaller set of supported tags, and add more later if/when really needed. Even links are debatable, why would you want to put links there, if there is a separate `websiteUrl`?

[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/shared/markdown.dart:98]: Maybe `<br>` support would be something to add early on...

[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/shared/markdown.dart]: Can we at-least add `<code>`, `<u>` and `<del>`, those seem harmless to me.

File: app/lib/shared/markdown.dart [hunk 3]
```diff
@@ -81,7 +98,24 @@ String _renderSafeHtml(List<m.Node> nodes) {
       return _whitelistedClassNames.contains(cn);
     },
   );
-  return html + '\n';
+  return inlineOnly
+      ? html.replaceAll(_multiLineBreakRegExp, '<br />\n')
```
Comment (jonasfj) [inline app/lib/shared/markdown.dart:102]: Can we please do this before the call to `_keepOnlyInlineElements` and then explicitly allow `<br>`... that way we don't ever output HTML that didn't go through some variant of `sanitizeHtml`.

My point: if hacking HTML manually please do so before any calls to `sanitizeHtml`.

**Pull request merged**